.circleci/regenerate.py

```diff
@@ -87,18 +92,19 @@ def build_workflow_pair(btype, os_type, python_version, cu_version, filter_branc
     w = []
     base_workflow_name = f"{prefix}binary_{os_type}_{btype}_py{python_version}_{cu_version}"
     w.append(generate_base_workflow(base_workflow_name, python_version, cu_version, filter_branch, os_type, btype))
-
-    if upload:
-
-        w.append(generate_upload_workflow(base_workflow_name, filter_branch, os_type, btype, cu_version))
-
-    if os_type != "macos":
-        pydistro = "pip" if btype == "wheel" else "conda"
-        w.append(
-            generate_smoketest_workflow(
-                pydistro, base_workflow_name, filter_branch, python_version, cu_version, os_type
+    skip_workflow = os_type == "linux" and btype == "conda" and python_version == "3.8" and cu_version == "cu116"
```

Comment:
It seems that moving this line to the call site and merge it into `upload` variable would make the logic simpler.